stdlib/asm/crypto/hashes/sha256.masm

```diff
@@ -0,0 +1,3096 @@
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.0
+    repeat.3
+        swap
+        drop
+    end
+
+    popw.mem # write to mem msg[0, 1, 2, 3] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+    
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.0
+    drop
+    repeat.2
+        swap
+        drop
+    end
+
+    popw.mem # write to mem msg[4, 5, 6, 7] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.0
+    drop
+    drop
+    swap
+    drop
+
+    popw.mem # write to mem msg[8, 9, 10, 11] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.0
+    drop
+    drop
+    drop
+
+    popw.mem # write to mem msg[12, 13, 14, 15] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.1
+    repeat.3
+        swap
+        drop
+    end
+
+    popw.mem # write to mem msg[16, 17, 18, 19] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.1
+    drop
+    repeat.2
+        swap
+        drop
+    end
+
+    popw.mem # write to mem msg[20, 21, 22, 23] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.1
+    drop
+    drop
+    swap
+    drop
+
+    popw.mem # write to mem msg[24, 25, 26, 27] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.1
+    drop
+    drop
+    drop
+
+    popw.mem # write to mem msg[28, 29, 30, 31] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.2
+    repeat.3
+        swap
+        drop
+    end
+
+    popw.mem # write to mem msg[32, 33, 34, 35] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.2
+    drop
+    repeat.2
+        swap
+        drop
+    end
+
+    popw.mem # write to mem msg[36, 37, 38, 39] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.2
+    drop
+    drop
+    swap
+    drop
+
+    popw.mem # write to mem msg[40, 41, 42, 43] #
+    pushw.local.4
+
+    exec.reorder_stack_words
+
+    # --- #
+
+    exec.gen_four_message_words
+
+    popw.local.4
+    movupw.2
+
+    pushw.local.2
+    drop
+    drop
+    drop
+
+    popw.mem # write to mem msg[44, 45, 46, 47] #
+    pushw.local.4
```

Comment:
This could be replaced by something like:
```
storew.mem
loadw.local.4
```
This change should save 8 cycles.

Reply:
Noted